docs/docs/guides/web3_plugin_guide/plugin_authors.md

```diff
@@ -210,3 +200,177 @@ public link(parentContext: Web3Context) {
+
+:::warning
+By augmenting `Web3Context` (and by extension all class interfaces that extend it), your plugin's interface will show up in things like intellisense for **all** Web3 modules that extend `Web3Context`, even if your plugin isn't registered - This is something worth making your users aware of, as they'll only be able to use your plugin if they register it with a Web3 class instance using `.registerPlugin`
+
+For context, here is a code example of this augmentation side-effect (the code in this example is further explained in the subsequent sections):
+
+```typescript
+// custom_rpc_methods_plugin.ts
+import { Web3PluginBase } from 'web3-core';
+
+import { Web3Context } from './reexported_web3_context';
+
+export class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods';
+
+	public someMethod() {
+		return 'someValue';
+	}
+}
+
+// Module Augmentation
+declare module './reexported_web3_context' {
+	interface Web3Context {
+		customRpcMethods: CustomRpcMethodsPlugin;
+	}
+}
+
+export { Web3Context };
+```
+
+The following represent what your plugin users would see:
+
+![web3 context augmentation side effect](./assets/web3_context_augmentation_side_effect.png 'web3Context augmentation side effect')
+
+The above screenshot shows intellisense thinking `.customRpcMethods.someMethod` is available to call on the instance of `Web3`, even though we haven't registered `CustomRpcMethodsPlugin` - running this code would result in an error.
+:::
+
+### Re-exporting Web3Context
+
+Currently TypeScript's module augmentation only supports named exports, so the first step in augmenting `Web3Context` is to re-export it as a named export. To do this we're going to create a `reexported_web3_context.ts` file (the name of this file can be whatever, but for the sake of this guide, we're going to assume it's named `reexported_web3_context.ts` and is located within the same directory as our `custom_rpc_methods_plugin.ts` file). The file contents should be as follows:
+
+```typescript
+import { Web3Context } from 'web3-core';
+
+export { Web3Context };
+```
+
+### Re-declaring the Module
+
+Now we're going to tell TypeScript that we're interested in re-defining a module's (in this case `reexported_web3_context`) interface. In simpler terms, TypeScript is already aware of what methods and classes exist for each web3.js module, but when registering a plugin, we're adding additional methods and/or classes to the module's interface and TypeScript needs a little help understanding what's going to be available within the module after the plugin is registered.
+
+We start with the following:
```

Comment:
I think it is not the "start" here. I think it is the main logic or the main file that contains the plugin class.

```suggestion
Here is the content of the source file that contains the main plugin class:
```
(This would also remove "we" to keep with the context)

Reply:
Addressed bu [this](https://github.com/web3/web3.js/pull/5621/commits/4cb635cb4a8489aaf3d0cd3b6479aad3764ed20e) commit

Reply:
Addressed by [this](https://github.com/web3/web3.js/pull/5621/commits/4cb635cb4a8489aaf3d0cd3b6479aad3764ed20e) commit